File: DSA Using Java/Exercise/BinaryToDecimal.java
public class BinaryToDecimal {
    public static int getDecimal(int n){
        int decimal = 0;
        int power = 0;

        while (n > 0) {
            int lastDigit = n % 10;
            lastDigit = (int) (lastDigit * Math.pow(2, power));
            decimal += lastDigit;
            n /= 10;
            power++;
        }
        return decimal;
    }
    public static void main(String[] args) {
        int decimal = getDecimal(100);

        System.out.println("Decimal = "+decimal);
    }
}

File: DSA Using C++/Basics of DSA/Binary Number System/binary_to_decimal.cpp
#include <iostream>

using namespace std;

int binaryToDecimal(string str)
{
    int decimal = 0, pow = 1;
    for (int i = str.length() - 1; i >= 0; i--)
    {
        int lastDigit = str[i] - '0';
        decimal += (lastDigit * pow);
        pow = pow << 1; // pow *= 2
    }

    return decimal;
}

int main()
{
    string binary;
    cout << "Enter a binary number: ";
    cin >> binary;

    cout << "Decimal = " << binaryToDecimal(binary) << endl;

    return 0;
}